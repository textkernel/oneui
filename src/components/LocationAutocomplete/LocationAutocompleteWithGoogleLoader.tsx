import * as React from 'react';
import { LoadScriptNext, LoadScriptNextProps } from '@react-google-maps/api';
import { LoadingSpinner } from '../LoadingSpinner';
import { LocationAutocomplete, LocationAutocompleteProps } from './LocationAutocomplete';

interface Props extends LocationAutocompleteProps {
    /** Google API key */
    apiKey: string;
    /**
     * The language code to be used for the map (e.g en). By default the users browser language will be used
     * For available values see: https://developers.google.com/maps/faq#languagesupport
     */
    language?: string;
    /** Regional setting for the map. By default Google uses US.
     * For details see: https://developers.google.com/maps/documentation/javascript/localization#Region
     */
    region?: string;
    /** other props to pass to the google loader. For details see: https://react-google-maps-api-docs.netlify.com/#loadscriptnext */
    additionalGoogleProps?: Omit<LoadScriptNextProps, 'googleMapsApiKey'>;
}

// for details see: https://developers.google.com/maps/documentation/javascript/libraries
const GOOGLE_API_LIBRARIES: (
    | 'drawing'
    | 'geometry'
    | 'localContext'
    | 'places'
    | 'visualization'
)[] = ['places'];

const LocationAutocompleteWithGoogleLoader: React.FC<Props> = ({
    apiKey,
    language = undefined,
    region = undefined,
    additionalGoogleProps = {},
    ...rest
}) => {
    return (
        <LoadScriptNext
            googleMapsApiKey={apiKey}
            language={language}
            region={region}
            loadingElement={<LoadingSpinner centerIn="parent" />}
            libraries={GOOGLE_API_LIBRARIES}
            {...additionalGoogleProps}
        >
            <LocationAutocomplete {...rest} />
        </LoadScriptNext>
    );
};

LocationAutocompleteWithGoogleLoader.displayName = 'LocationAutocompleteWithGoogleLoader';

export { LocationAutocompleteWithGoogleLoader, Props as LocationAutocompleteWithGoogleLoaderProps };
