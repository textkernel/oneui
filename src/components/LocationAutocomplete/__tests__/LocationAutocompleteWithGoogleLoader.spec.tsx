import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { LocationAutocompleteWithGoogleLoader } from '../LocationAutocompleteWithGoogleLoader';

describe('<LocationAutocompleteWithGoogleLoader/> that loads google api and renders a LocationAutocompleteWithGoogleLoader', () => {
    it('should render with default props', () => {
        const view = render(
            <LocationAutocompleteWithGoogleLoader
                apiKey="someKey"
                inputPlaceholder="Location..."
                noSuggestionsPlaceholder="No suggestions..."
                onSelectionChange={() => null}
            />
        );

        expect(view.container).toMatchSnapshot();
        expect(screen.getByRole('status')).toBeVisible();
    });
});
