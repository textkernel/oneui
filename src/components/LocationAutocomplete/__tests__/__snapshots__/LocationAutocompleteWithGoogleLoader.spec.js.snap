// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationAutocompleteWithGoogleLoader/> that loads google api and renders a LocationAutocompleteWithGoogleLoader should render with default props 1`] = `
<LocationAutocompleteWithGoogleLoader
  apiKey="someKey"
  inputPlaceholder="Location..."
  noSuggestionsPlaceholder="No suggestions..."
  onSelectionChange={[Function]}
>
  <Memo(LoadScriptNext)
    googleMapsApiKey="someKey"
    libraries={
      Array [
        "places",
      ]
    }
    loadingElement={
      <LoadingSpinner
        centerIn="parent"
      />
    }
  >
    <LoadingSpinner
      centerIn="parent"
    >
      <div
        aria-busy={true}
        aria-hidden={false}
        className="LoadingSpinner LoadingSpinner--centerIn_parent"
        role="status"
      >
        <svg
          className="LoadingSpinner__svg"
          style={null}
          viewBox={
            Array [
              0,
              0,
              44,
              44,
            ]
          }
        >
          <circle
            className="LoadingSpinner__path LoadingSpinner__path--context_brand"
            cx="22"
            cy="22"
            fill="none"
            r="20"
            strokeWidth="4"
          />
        </svg>
      </div>
    </LoadingSpinner>
  </Memo(LoadScriptNext)>
</LocationAutocompleteWithGoogleLoader>
`;
